capellambse/filehandler/abc.py

```diff
@@ -251,6 +252,25 @@ def iterdir(
     def name(self) -> str:
         return self._path.name
 
+    @property
+    def suffix(self) -> str:
+        return self._path.suffix
+
+    @property
+    def stem(self) -> str:
+        return self._path.stem
+
+    @property
+    def parent(self):
+        return self._path.parent
```

Comment:
`self._path.parent` is an AbstractPosixPath, which would need to be turned into an instance of `type(self)` before returning it.

Also missing return annotation: `(...) -> te.Self:`